### PR TITLE
Fixed /scim/v2/User/# endpoint to return 404 instead of 500

### PR DESCRIPTION
**Pull request by QuanMPhm**
Fixes nerc-project/coldfront-plugin-api#15 by performing a try-except block around the query call.

### CODE REVIEW

**Comment by knikolla** [comment]
The code looks fine, but your PRs are trying to pull in a large number of unrelated commits. So far I've been squashing and merging your PRs but this is a good opportunity for you to learn some intermediate git operations like rebasing and squashing. I'll approve once this PR is 1 commit only.

File: src/coldfront_plugin_api/scim_v2/users.py [hunk 1]
```diff
@@ -108,5 +109,8 @@ class UserDetail(APIView):
     permission_classes = [IsAdminUser]
 
     def get(self, request, username, format=None):
-        allocation = User.objects.get(username=username)
+        try:
+            allocation = User.objects.get(username=username)
+        except User.DoesNotExist:
+            raise Http404("Username not found")
```
Comment (knikolla) [inline src/coldfront_plugin_api/scim_v2/users.py:115]: Does this return JSON or HTML?

Comment (QuanMPhm) [reply src/coldfront_plugin_api/scim_v2/users.py:115]: Entering an invalid username returns a HTML page that seems to be part of Django REST framework. It seems like it doesn't matter what I put as the input string to Http404, the REST framework is overriding whatever Django's Http404 page is.

Comment (knikolla) [reply src/coldfront_plugin_api/scim_v2/users.py:115]: You don’t need to use Http404. There are other places in this file where I manually return a response with a specific status code.

**Pull request closed without merging**